src/reconcile/rec_visibility.c

```diff
@@ -463,7 +463,9 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
      */
     if (__rec_need_save_upd(
           session, upd_select->upd, max_txn, max_ts, list_uncommitted, r->flags)) {
-        WT_ASSERT(session, r->max_txn != WT_TS_NONE);
+        /* During recovery, there are no transaction id's. */
+        if (!F_ISSET(S2C(session), WT_CONN_RECOVERING))
+            WT_ASSERT(session, r->max_txn != WT_TS_NONE);
```

Comment:
You might get dinged by static analyzers that there's an `if` without any content because `WT_ASSERT` is only "real" in diagnostic mode. I'd go with:
```
WT_ASSERT(session, !F_ISSET(S2C(session), WT_CONN_RECOVERING) || r->max_txn != WT_TS_NONE);
```

Reply:
Fixed as you suggested.